test/Smt.test.ts

```diff
@@ -0,0 +1,87 @@
+// @ts-ignore
+import * as snarkjs from "snarkjs";
+
+import { expect } from "chai";
+
+import { SignerWithAddress } from "@nomicfoundation/hardhat-ethers/signers";
+import { CircomJS } from "@zefi/circomjs";
+import { ethers } from "hardhat";
+import { deployPoseidonFacade } from "./helpers/poseidon/poseidon-deployer";
+import { Reverter } from "./helpers/reverter";
+
+import {
+  SmtChecker,
+  SmtChecker__factory,
+  SmtVerifier,
+} from "../typechain-types";
+
+import { Calldata, ProofStruct } from "./helpers/types";
+
+describe("SMT test", () => {
+  const reverter = new Reverter();
+  const circom = new CircomJS();
+
+  const smtCircuit = circom.getCircuit("smt");
+
+  let smtChecker: SmtChecker;
+  let smtVerifier: SmtVerifier;
+  let owner: SignerWithAddress;
+
+  beforeEach("setup", async () => {
+    [owner] = await ethers.getSigners();
+
+    const poseidonFacade = await deployPoseidonFacade();
+
+    smtVerifier = await ethers.deployContract("SmtVerifier");
+
+    const SMTChecker = new SmtChecker__factory({
+      ["@iden3/contracts/lib/Poseidon.sol:PoseidonFacade"]:
+        await poseidonFacade.getAddress(),
+    });
+
+    smtChecker = await SMTChecker.connect(owner).deploy(
+      await smtVerifier.getAddress()
+    );
+
+    reverter.snapshot();
+  });
+
+  afterEach(async () => {
+    await reverter.revert();
+  });
+
+  it("should prove the tree", async () => {
+    let leaves: string[] = [];
+
+    for (let i = 0; i < 10; i++) {
```

Comment:
Cool, but we need more tests

Reply:
Will be done by Ruslan as he already works on SMT tree. I will take IMT